runtime/lua/vim/treesitter/dev.lua

```diff
@@ -99,22 +99,44 @@ function TSTreeView:new(bufnr, lang)
   -- For each child tree (injected language), find the root of the tree and locate the node within
   -- the primary tree that contains that root. Add a mapping from the node in the primary tree to
   -- the root in the child tree to the {injections} table.
-  local root = parser:parse(true)[1]:root()
-  local injections = {} ---@type table<integer,table>
-  for _, child in pairs(parser:children()) do
-    child:for_each_tree(function(tree, ltree)
-      local r = tree:root()
-      local node = root:named_descendant_for_range(r:range())
-      if node then
-        injections[node:id()] = {
-          lang = ltree:lang(),
-          root = r,
-        }
-      end
-    end)
+  local tstrees = parser:parse(true)
+  local injections = {} ---@type table<integer,{lang: string, root: TSNode}>
+
+  ---@param node TSNode
+  ---@return number
+  local function node_length(node)
+    local _, _, start_byte = node:start()
+    local _, _, end_byte = node:end_()
+    return end_byte - start_byte
   end
+  parser:for_each_tree(function(root_tree, root_ltree)
+    local root = root_tree:root()
+    for _, child in pairs(root_ltree:children()) do
+      child:for_each_tree(function(tree, ltree)
+        local r = tree:root()
+        local node = root:named_descendant_for_range(r:range())
+        if node then
+          if injections[node:id()] then
```

Comment:
I don't know if the inner injected trees are parsed before or after the outer ones. If the order are outer -> inner, then I will remove the if accordingly